Thanks for the PR, @dogeplusplus. Looks great in general; our CI for older versions seems to indicate that there's some incompatibility with ONNX 1.8.1 and 1.7.0:

```
tests/onnx/test_onnx_model_export.py::test_model_save_load FAILED        [  3%]
tests/onnx/test_onnx_model_export.py::test_model_save_load_2gb 
Error: Process completed with exit code 143.
```

Please let me know if you'd like assistance debugging this.

--- a/mlflow/onnx.py
+++ b/mlflow/onnx.py
@@ -217,9 +217,9 @@ def save_model(
 def _load_model(model_file):
     import onnx
 
+    onnx.checker.check_model(model_file)
     onnx_model = onnx.load(model_file)
     # Check Formation
-    onnx.checker.check_model(onnx_model)
     return onnx_model
 
 

--- a/mlflow/onnx.py
+++ b/mlflow/onnx.py
@@ -11,6 +11,7 @@ import os
 import yaml
 import numpy as np
 from pathlib import Path
+from packaging.version import Version
 
 import pandas as pd
 
@@ -157,7 +158,10 @@ def save_model(
     model_data_path = os.path.join(path, model_data_subpath)
 
     # Save onnx-model
-    onnx.save_model(onnx_model, model_data_path, save_as_external_data=True)
+    if Version(onnx.__version__) >= Version("1.9.0"):
+        onnx.save_model(onnx_model, model_data_path, save_as_external_data=True)
+    else:
+        onnx.save_model(onnx_model, model_data_path)
 
     pyfunc.add_to_model(
         mlflow_model,

--- a/tests/onnx/test_onnx_model_export.py
+++ b/tests/onnx/test_onnx_model_export.py
@@ -107,34 +107,6 @@ def onnx_model(model, sample_input, tmpdir):
     return onnx.load(model_path)
 
 
-@pytest.fixture
-def onnx_model_2gb(tmpdir):
-    model_path = os.path.join(str(tmpdir), "onnx_big")
-    dynamic_axes = {"input": {0: "batch"}}
-    # Linear layer weights + biases just over 1GB
-    big_model = nn.Sequential(
-        nn.Linear(2**14, 2**14),
-        nn.Linear(2**14, 2**14),
-    )
-
-    sample_input = torch.ones(1, 2**14)
-    if Version(torch.__version__) >= Version("1.11.0"):
-        torch.onnx.export(
-            big_model, sample_input, model_path, dynamic_axes=dynamic_axes, input_names=["input"]
-        )
-    # Exporting with pytorch use_external_data_format required for large models for <1.11
-    else:
-        torch.onnx.export(
-            big_model,
-            sample_input,
-            model_path,
-            dynamic_axes=dynamic_axes,
-            input_names=["input"],
-            use_external_data_format=True,
-        )
-    return onnx.load(model_path)
-
-
 @pytest.fixture(scope="module")
 def multi_tensor_model(dataset):
     class MyModel(nn.Module):
@@ -265,7 +237,14 @@ def onnx_custom_env(tmpdir):
 
 
 def test_model_save_load(onnx_model, model_path):
+    # New ONNX versions can optionally convert to external data
+    if Version(onnx.__version__) >= Version("1.9.0"):
+        onnx.convert_model_to_external_data = mock.Mock()
+
     mlflow.onnx.save_model(onnx_model, model_path)
+
+    if Version(onnx.__version__) >= Version("1.9.0"):
+        assert onnx.convert_model_to_external_data.called
 
     # Loading ONNX model
     onnx.checker.check_model = mock.Mock()
@@ -273,13 +252,20 @@ def test_model_save_load(onnx_model, model_path):
     assert onnx.checker.check_model.called
 
 
-def test_model_save_load_2gb(onnx_model_2gb, model_path):
-    mlflow.onnx.save_model(onnx_model_2gb, model_path)
+def test_model_save_load_nonexternal_data(onnx_model, model_path):
+    original_save_model = onnx.save_model
+    if Version(onnx.__version__) >= Version("1.9.0"):
 
-    # Loading >2GB ONNX model
-    onnx.checker.check_model = mock.Mock()
-    mlflow.onnx.load_model(model_path)
-    assert onnx.checker.check_model.called
+        def onnx_save_nonexternal(model, path, save_as_external_data):
+            original_save_model(model, path, save_as_external_data=False)
+
+        with mock.patch("onnx.save_model", wraps=onnx_save_nonexternal):
+            mlflow.onnx.save_model(onnx_model, model_path)
+
+        # Loading ONNX model
+        onnx.checker.check_model = mock.Mock()
+        mlflow.onnx.load_model(model_path)
+        assert onnx.checker.check_model.called
 
 
 def test_signature_and_examples_are_saved_correctly(onnx_model, data, onnx_custom_env):

